java/src/main/java/org/whispersystems/signalservice/loki/api/LokiDotNetAPI.kt

```diff
@@ -237,6 +237,68 @@ open class LokiDotNetAPI(private val userHexEncodedPublicKey: String, private va
             throw PushNetworkException(exception)
         }
     }
+
+    // upload profile photo
+    // refer to the upload function above
+    fun uploadProfilePhoto(server: String, profilePhoto: ByteArray): Triple<Long, String, ByteArray> {
```

Comment:
you can do the following
```
val inputStream = ByteArrayInputStream(profilePhoto)
return upload(server, inputStream, "application/binary", profilePhoto.length, BasicOutputStreamFactory(), null)
```

`BasicOutputStreamFactory` will just copy the input stream directly to the output stream.

Also i think we might need to modify the endpoint depending upon what we're uploading.